package/MDAnalysis/analysis/rms.py

```diff
@@ -381,11 +394,15 @@ def __init__(self, atomgroup, reference=None, select='all',
 
             .. Note:: Experimental feature. Only limited error checking
                       implemented.
-        weights : {"mass", ``None``} or array_like (optional)
-             choose weights. With ``"mass"`` uses masses as weights; with ``None``
-             weigh each atom equally. If a float array of the same length as
+        weights : {"mass", ``None``} or array_like or list of array_like (optional)
+             choose weights. With ``"mass"`` uses masses as weights for both `select`
+             and `groupselections`; with ``None`` weigh each atom equally for both 
+             `select` and `groupselections`. If a float array of the same length as
              `atomgroup` is provided, use each element of the `array_like` as a
-             weight for the corresponding atom in `atomgroup`.
+             weight for the corresponding atom in `select`, and assume ``None`` 
+             for `groupselections`. 
+             If weights is a list or array_like or 'mass' or None of length 
+             (1 + length(`groupselections`)), apply the weights correspondingly.
```

Comment:
For me this is a bit confusing. Couldn't we split this into two weight attributes? One for `select` and one for `groupselections`. I'm not an expert for RMSDs but this reads like it is too much for a single attribute.

Reply:
I agree it becomes quite confusing/complicated (also for the code) after the change. But from a normal user's perspective, one normally uses `None`––which is the default for all selections; one might use `mass` in some rare cases––if splitting ´weight´ into two, you have to manually set both into `mass` (which is kind of prone to error?); I might never use custom weight...

Reply:
I agree with it might be better to split them. You could set it up to mimic the current behaviour - i.e. if weights aren't provided for `groupselections`, values are set based on the value provided for `select` rather than using a default.